liquidctl/driver/commander_core.py

```diff
@@ -51,7 +51,7 @@ class CommanderCore(UsbHidDriver):
     """Corsair Commander Core"""
 
     SUPPORTED_DEVICES = [
-        (0x1b1c, 0x0c1c, None, 'Corsair Commander Core (experimental)', {})
```

Comment:
Naturally you'll need to keep the old Core entry and add a new one for the Core XT.

---

Otherwise, I'm glad to hear that `initialize()` appears to be working.  What are the physical differences between the Core and Core XT devices?

Reply:
i can throw up pictures in a few minutes

Reply:
a quick description: it takes 5V from its SATA hookup for logic, and presumably also 12V for the fans (have not verified this last, but it definitely needs 5V). it has a usb2 female which wants a male header. USB2 only gives you 500mA of 5V, so the SATA connector's probably necessary just for 5V; this doesn't tell us whether it's pulling down off the 12V lines. both must be connected for the device to work. it then has 6 "RGB Hub" proprietary connections, 6 "Fan" 4-pin PWM connections, 2 temperature sensing inputs, an "RGB" port for driving the proprietary corsair protocol, and the mysterious "ID".


```
Bus 009 Device 010: ID 1b1c:0c2a Corsair CORSAIR iCUE COMMANDER CORE XT
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1b1c Corsair
  idProduct          0x0c2a 
  bcdDevice            1.00
  iManufacturer           1 Corsair
  iProduct                2 CORSAIR iCUE COMMANDER CORE XT
  iSerial                 3 d10600908ac184aacaa70c064091005f
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0042
    bNumInterfaces          2
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xc0
      Self Powered
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.00
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      34
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x84  EP 4 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0300  1x 768 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x04  EP 4 OUT
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0400  1x 1024 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.00
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      21
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0300  1x 768 bytes
        bInterval               1
Device Qualifier (for other device speed):
  bLength                10
  bDescriptorType         6
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  bNumConfigurations      0
can't get debug descriptor: Resource temporarily unavailable
Device Status:     0x0001
  Self Powered
```